sql/src/main/java/com/rilldata/SqlConverterEntrypoint.java

```diff
@@ -44,31 +44,20 @@ public static CCharPointer convertSql(IsolateThread thread, AllocatorFn allocato
     }
   }
 
-  @CEntryPoint(name = "create_source")
-  public static byte[] createSource(IsolateThread thread, CCharPointer sourceDef, CCharPointer schema)
-  {
-    try {
-      String javaSchemaString = CTypeConversion.toJavaString(schema);
-      SqlConverter sqlConverter = new SqlConverter(javaSchemaString);
-      String createSource = CTypeConversion.toJavaString(sourceDef);
-      return sqlConverter.createSource(createSource);
-    } catch (Exception e) {
-      e.printStackTrace();
-      return null;
-    }
-  }
-
-  @CEntryPoint(name = "create_metrics_view")
-  public static byte[] createMetricsView(IsolateThread thread, CCharPointer metricsViewDef, CCharPointer schema)
+  @CEntryPoint(name = "get_ast")
+  public static CCharPointer getAST(IsolateThread thread, AllocatorFn allocatorFn, CCharPointer sql,
+      CCharPointer schema
+  )
   {
     try {
       String javaSchemaString = CTypeConversion.toJavaString(schema);
       SqlConverter sqlConverter = new SqlConverter(javaSchemaString);
-      String createMetricsView = CTypeConversion.toJavaString(metricsViewDef);
-      return sqlConverter.createMetricsView(createMetricsView);
+      String sqlString = CTypeConversion.toJavaString(sql);
+      byte[] ast = sqlConverter.getAST(sqlString);
+      return convertToCCharArrayPointer(allocatorFn, ast);
```

Comment:
it should work with `convertToCCharPointer`

Reply:
changed